carla_ros_bridge/src/carla_ros_bridge/actor_factory.py

```diff
@@ -351,6 +352,16 @@ def _create_object(self, uid, type_id, name, attach_to, spawn_pose, carla_actor=
                                  parent=parent,
                                  node=self.node)
 
+        elif type_id == EgoTrafficLightSensor.get_blueprint_name():
+            actor = EgoTrafficLightSensor(
+                uid=uid,
+                name=name,
+                parent=parent,
+                node=self.node,
+                actor_list=self.actors,
+                carla_map=self.world.get_map()
```

Comment:
Since get map is expensive, does it cause lag here? Can we pre-load the map?

Reply:
if you arent seeing lag we can consider not doing that, just curious.

Reply:
I didn't notice any lag. Now when reading Carla APi documentation, it looks like this calls to get the current XODR map. I think if we store it, then when a new map is loaded, I don't think the reference will have latest map. I will check it.

`
self. map = world.get_map()
# When a map is changed, self.map will have the old value
`

Reply:
As I expected, when call `world.load_world()`, the old `self.map` is not updated. We need to call self.world.get_map() again.

That means the logic I have for checking if the map has been changed won't work.

Is there a way to detect map change? I tried listen to this topic: `/carla/world_info` but I don't see it outputs anything when the map changes either.

Reply:
I need the map object to get the waypoints that ego is currently on. Get map is called once at beginning or when a new map is loaded so I think it should be fine.

Regarding detecting if the map has changed. It will require passing the CarlaClient to ActorFactory, then pass it to whatever sensors we want to detect the map change. The reason are:
- carla_world.id. This one doesn't change when load a new map
- carla_client.get_world().id. This one does.

I think if we want to load a new map maybe it's better to rerun the motion sim with the new map at this point. That will make sure the rosbag correct and clean; especially, other sensors in carla rosbridge don't detect loading a new map.

Maybe hold off on merging this and let's try it on the motion platform to see if there are any delay then?

Reply:
fyi @anrp-tri